Port forward license data from maintenance (KEYCLOAK-4950), with improvements.

ASzc opened the pull request:
org.keycloak dependencies will be automatically added to the xml during the build, removing the need for runs of download-license-files.sh every time the keycloak version changes.

Documentation on "why and how" for the license data has also been added.

To reduce code duplication issues, plugin definitions are stored in keycloak-parent, but only active in the projects that need them (not bound to any phase by default). Also, the common files have been moved into licenses/common/, so that a single symlink will suffice to replicate the current and future files needed by the plugin executions. While the assembly.xml definitions remain duplicated, they are fairly minimal and shouldn't need to change often.

License data is available for all adapters shipped in the product, plus server-feature-pack.

The keycloak slot is populated with data, in addition to the rh-sso slot. A number of the adapters don't depend on any third-party artifacts, so they have (mostly) blank license.xml files.

ASzc committed:
Port forward license data from maintenance, with improvements.

org.keycloak dependencies will be automatically added to the xml during the
build, removing the need for runs of download-license-files.sh every time the
keycloak version changes.

Documentation on "why and how" for the license data has also been added.

To reduce code duplication issues, plugin definitions are stored in
keycloak-parent, but only active in the projects that need them (not bound to
any phase by default). Also, the common files have been moved into
licenses/common/, so that a single symlink will suffice to replicate the
current and future files needed by the plugin executions. While the
assembly.xml definitions remain duplicated, they are fairly minimal and
shouldn't need to change often.

License data is available for all adapters shipped in the product, plus
server-feature-pack.

The keycloak slot is populated with data, in addition to the rh-sso slot. A
number of the adapters don't depend on any third-party artifacts, so they have
(mostly) blank license.xml files.

ASzc committed:
Explicitly define the dependency for unpacking in js-adapter-zip.

This makes it visible to append-product-licenses.groovy and other plugins.

ASzc committed:
Initial run of download-license-files.sh on license XMLs

Command line used:

find -path '*/src/main/resources/licenses' | while read -r r; do pushd "$r"; common/download-license-files.sh rh-sso/licenses.xml; common/download-license-files.sh keycloak/licenses.xml; popd; done

stianst commented:
There's two copies of download-license-files.sh currently, but then there are many more licenses.xml files. Would be better to download-license-files.sh in one place, maybe misc/scripts or something?

When running the download-license-files.sh against the currently checked in licenses.xml files I would assume there would be no changes needed. However that's not the case. After running it against all licenses.xml files I end up with 97 changed files. https://github.com/stianst/keycloak/commit/3773ff68eb70d422b21c8ab109c03ee506304ff0 shows what I ended up with after running this.

Files should only change if the relevant license information has changed otherwise we'll end up with very messy commits.

stianst commented:
I also tried to edit the version of one single dependency. This can be seen in https://github.com/stianst/keycloak/commit/ee24906a5f4e1807238d327ea56ebb8650c79688. The changes doesn't make sense to me. I changed version for com.google.zxing:javase which resulted in license file for it being "renamed" (well really deleted, then added, but Git figures out it's a rename). However, it's also changing the file "com.google.zxing,core,3.2.1,Apache Software License 2.0.txt" which looks like it's incorrect as it looks like it has the wrong artifact name since it's javase not core as you'd expect from the filename.

ASzc commented:
@stianst I only see one download-license-files.sh, located here:

```
./distribution/feature-packs/server-feature-pack/src/main/resources/licenses/common/download-license-files.sh
```
There are other virtual copies of the file created by the directory symlinks for `licenses/common`:
```
./distribution/saml-adapters/as7-eap6-adapter/eap6-adapter-zip/src/main/resources/licenses/common/download-license-files.sh
./distribution/saml-adapters/wildfly-adapter/wildfly-adapter-zip/src/main/resources/licenses/common/download-license-files.sh
./distribution/adapters/as7-eap6-adapter/eap6-adapter-zip/src/main/resources/licenses/common/download-license-files.sh
./distribution/adapters/wildfly-adapter/src/main/resources/licenses/common/download-license-files.sh
./distribution/adapters/js-adapter-zip/src/main/resources/licenses/common/download-license-files.sh
./distribution/adapters/fuse-adapter-zip/src/main/resources/licenses/common/download-license-files.sh
```

The excessive commit diff thing is probably due to system-dependent sorting (the deduplication sort order is different on your system to mine), possibly from a different locale or maybe different distro defaults. I can fix that I think.

I would expect the license files for `com.google.zxing:javase` and `com.google.zxing:core` to be symlinked together, as they're from the same project, and so likely pull identical license text. Indeed, that's what I'm seeing in your commit. `javase`'s filename changed (new version), and git displays the updated symlink from `core` to `javase`'s new filename. I wish Github would display symlinks better.

ASzc committed:
Enforce a standard sort order on deduplication in download-license-files.sh

This will avoid noise in the diffs (files switching between symlink and regular
status) when users on systems with a different default sort order run the
script. `LC_ALL=C sort` will sort by byte order.

ASzc commented:
Added a commit to the PR that should fix the noise issue, relative to that commit (I re-ran the script too, and some of the license files changed with the new sorting). It enforces byte sort order, so it will be consistent between systems. It will do odd things with any unicode characters, but the actual sort order doesn't matter, provided it's consistent.

stianst commented:
Actually now there's more changes:
https://github.com/stianst/keycloak/commit/1b00f2e012500622d27555e8e5a1b496d63632e5

stianst commented:
I finally get your point on how GitHub shows symlinks. That sure isn't very nice. Doesn't seem to be any hints that it is a symlink, just looks like a "weird" file.

ASzc commented:
Ah, I forgot to put `sort` into null line-ending mode, so it's not actually sorting. I'll be force-pushing an update to the last commit in a few minutes.

stianst commented:
Less, but still a lot of changes:
https://github.com/stianst/keycloak/commit/20b3bac9f1e18ec1853c582012d9a1086a59adf8

Can you maybe try this on a Docker image or something so that you can reproduce this on your end? I can't keep reviewing this again and again.

ASzc commented:
Sorry. That's really baffling now. You use Ubuntu right? I'll try docker, but if it's tied to the filesystem type I don't think that will be enough.

ASzc commented:
Unable to reproduce with Docker (I'll see if I can get an Ubuntu VM going...), here's what I ran:
```
[aszczucz@tyrfing ~]$ docker run --rm -ti ubuntu:latest
root@ba09ed4de475:/keycloak# grep RELEASE /etc/*release
/etc/lsb-release:DISTRIB_RELEASE=16.04
root@ba09ed4de475:/# apt-get update
root@ba09ed4de475:/# apt-get install git xmlstarlet dos2unix curl
root@ba09ed4de475:/# git clone --branch KEYCLOAK-4950_master https://github.com/ASzc/keycloak.git
root@ba09ed4de475:/# cd keycloak
root@ba09ed4de475:/keycloak# find -path '*/src/main/resources/licenses' | while read -r r; do pushd "$r"; common/download-license-files.sh rh-sso/licenses.xml; common/download-license-files.sh keycloak/licenses.xml; popd; done
/keycloak/distribution/adapters/wildfly-adapter/src/main/resources/licenses /keycloak
==> Deleting old license files in rh-sso
==> Munging rh-sso/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for rh-sso/licenses.xml into rh-sso
==> Normalizing license line endings
==> Symlinking identical files
==> Complete
==> Deleting old license files in keycloak
==> Munging keycloak/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for keycloak/licenses.xml into keycloak
==> Normalizing license line endings
==> Symlinking identical files
==> Complete
/keycloak
/keycloak/distribution/adapters/fuse-adapter-zip/src/main/resources/licenses /keycloak
==> Deleting old license files in rh-sso
==> Munging rh-sso/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for rh-sso/licenses.xml into rh-sso
org.bouncycastle,bcpkix-jdk15on,1.56.0.redhat-2,MIT License.txt
org.bouncycastle,bcprov-jdk15on,1.56.0.redhat-2,MIT License.txt
com.fasterxml.jackson.core,jackson-annotations,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4.redhat-1,Apache Software License 2.0.txt
commons-codec,commons-codec,1.9,Apache Software License 2.0.txt
commons-logging,commons-logging,1.2,Apache Software License 2.0.txt
org.apache.httpcomponents,httpclient,4.5.0.redhat-1,Apache Software License 2.0.txt
org.apache.httpcomponents,httpcore,4.4.1.redhat-1,Apache Software License 2.0.txt
org.jboss.logging,jboss-logging,3.3.0.Final-redhat-1,Apache Software License 2.0.txt
==> Normalizing license line endings
dos2unix: converting file rh-sso/com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.bouncycastle,bcprov-jdk15on,1.56.0.redhat-2,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.core,jackson-core,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/commons-logging,commons-logging,1.2,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.bouncycastle,bcpkix-jdk15on,1.56.0.redhat-2,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.core,jackson-databind,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.httpcomponents,httpcore,4.4.1.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/commons-codec,commons-codec,1.9,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.core,jackson-annotations,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.httpcomponents,httpclient,4.5.0.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.jboss.logging,jboss-logging,3.3.0.Final-redhat-1,Apache Software License 2.0.txt to Unix format ...
==> Symlinking identical files
com.fasterxml.jackson.core,jackson-annotations,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4.redhat-1,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4.redhat-1,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4.redhat-1,Apache Software License 2.0.txt
commons-codec,commons-codec,1.9,Apache Software License 2.0.txt
 -> commons-logging,commons-logging,1.2,Apache Software License 2.0.txt
 -> org.jboss.logging,jboss-logging,3.3.0.Final-redhat-1,Apache Software License 2.0.txt
org.apache.httpcomponents,httpclient,4.5.0.redhat-1,Apache Software License 2.0.txt
org.apache.httpcomponents,httpcore,4.4.1.redhat-1,Apache Software License 2.0.txt
org.bouncycastle,bcpkix-jdk15on,1.56.0.redhat-2,MIT License.txt
 -> org.bouncycastle,bcprov-jdk15on,1.56.0.redhat-2,MIT License.txt
==> Complete
==> Deleting old license files in keycloak
==> Munging keycloak/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for keycloak/licenses.xml into keycloak
org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt
org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt
com.fasterxml.jackson.core,jackson-annotations,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4,Apache Software License 2.0.txt
commons-codec,commons-codec,1.9,Apache Software License 2.0.txt
commons-logging,commons-logging,1.2,Apache Software License 2.0.txt
org.apache.httpcomponents,httpclient,4.5,Apache Software License 2.0.txt
org.apache.httpcomponents,httpcore,4.4.1,Apache Software License 2.0.txt
org.jboss.logging,jboss-logging,3.3.0.Final,Apache Software License 2.0.txt
==> Normalizing license line endings
dos2unix: converting file keycloak/org.jboss.logging,jboss-logging,3.3.0.Final,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/commons-logging,commons-logging,1.2,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/commons-codec,commons-codec,1.9,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.httpcomponents,httpcore,4.4.1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.httpcomponents,httpclient,4.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.core,jackson-databind,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.core,jackson-annotations,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.core,jackson-core,2.5.4,Apache Software License 2.0.txt to Unix format ...
==> Symlinking identical files
com.fasterxml.jackson.core,jackson-annotations,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4,Apache Software License 2.0.txt
commons-codec,commons-codec,1.9,Apache Software License 2.0.txt
 -> commons-logging,commons-logging,1.2,Apache Software License 2.0.txt
 -> org.jboss.logging,jboss-logging,3.3.0.Final,Apache Software License 2.0.txt
org.apache.httpcomponents,httpclient,4.5,Apache Software License 2.0.txt
org.apache.httpcomponents,httpcore,4.4.1,Apache Software License 2.0.txt
org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt
 -> org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt
==> Complete
/keycloak
/keycloak/distribution/adapters/js-adapter-zip/src/main/resources/licenses /keycloak
==> Deleting old license files in rh-sso
==> Munging rh-sso/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for rh-sso/licenses.xml into rh-sso
==> Normalizing license line endings
==> Symlinking identical files
==> Complete
==> Deleting old license files in keycloak
==> Munging keycloak/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for keycloak/licenses.xml into keycloak
==> Normalizing license line endings
==> Symlinking identical files
==> Complete
/keycloak
/keycloak/distribution/adapters/as7-eap6-adapter/eap6-adapter-zip/src/main/resources/licenses /keycloak
==> Deleting old license files in rh-sso
==> Munging rh-sso/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for rh-sso/licenses.xml into rh-sso
org.bouncycastle,bcpkix-jdk15on,1.52.0.redhat-3,MIT License.txt
org.bouncycastle,bcprov-jdk15on,1.52.0.redhat-3,MIT License.txt
com.fasterxml.jackson.core,jackson-annotations,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4.redhat-1,Apache Software License 2.0.txt
==> Normalizing license line endings
dos2unix: converting file rh-sso/org.bouncycastle,bcprov-jdk15on,1.52.0.redhat-3,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.core,jackson-core,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.core,jackson-databind,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.fasterxml.jackson.core,jackson-annotations,2.5.4.redhat-1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.bouncycastle,bcpkix-jdk15on,1.52.0.redhat-3,MIT License.txt to Unix format ...
==> Symlinking identical files
com.fasterxml.jackson.core,jackson-annotations,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4.redhat-1,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4.redhat-1,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4.redhat-1,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4.redhat-1,Apache Software License 2.0.txt
org.bouncycastle,bcpkix-jdk15on,1.52.0.redhat-3,MIT License.txt
 -> org.bouncycastle,bcprov-jdk15on,1.52.0.redhat-3,MIT License.txt
==> Complete
==> Deleting old license files in keycloak
==> Munging keycloak/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for keycloak/licenses.xml into keycloak
org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt
org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt
com.fasterxml.jackson.core,jackson-annotations,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4,Apache Software License 2.0.txt
==> Normalizing license line endings
dos2unix: converting file keycloak/com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.core,jackson-databind,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.core,jackson-annotations,2.5.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/com.fasterxml.jackson.core,jackson-core,2.5.4,Apache Software License 2.0.txt to Unix format ...
==> Symlinking identical files
com.fasterxml.jackson.core,jackson-annotations,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-core,2.5.4,Apache Software License 2.0.txt
com.fasterxml.jackson.core,jackson-databind,2.5.4,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-base,2.5.4,Apache Software License 2.0.txt
 -> com.fasterxml.jackson.jaxrs,jackson-jaxrs-json-provider,2.5.4,Apache Software License 2.0.txt
org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt
 -> org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt
==> Complete
/keycloak
/keycloak/distribution/feature-packs/server-feature-pack/src/main/resources/licenses /keycloak
==> Deleting old license files in rh-sso
==> Munging rh-sso/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for rh-sso/licenses.xml into rh-sso
com.google.zxing,core,3.2.1.redhat-4,Apache Software License 2.0.txt
com.google.zxing,javase,3.2.1.redhat-4,Apache Software License 2.0.txt
org.liquibase,liquibase-core,3.4.1.redhat-2,Apache Software License 2.0.txt
org.twitter4j,twitter4j-core,4.0.4.redhat-3,Apache Software License 2.0.txt
org.freemarker,freemarker,2.3.23.redhat-3,Apache Software License 2.0.txt
aopalliance,aopalliance,1.0,Public Domain.txt
com.google.code.guice,guice,2.0,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-classworlds,2.5.2,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-utils,3.0.20,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-component-annotations,1.5.5,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-interpolation,1.21,Apache Software License 2.0.txt
org.apache.ant,ant-launcher,1.8.3,Apache Software License 2.0.txt
org.apache.ant,ant,1.8.3,Apache Software License 2.0.txt
org.apache.maven.wagon,wagon-http-shared,2.6,Apache Software License 2.0.txt
org.apache.maven.wagon,wagon-provider-api,2.6,Apache Software License 2.0.txt
org.apache.maven.wagon,wagon-http,2.6,Apache Software License 2.0.txt
org.apache.maven,maven-compat,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-model-builder,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-core,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-model,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-artifact,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-plugin-api,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-settings,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-repository-metadata,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-aether-provider,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-settings-builder,3.2.5,Apache Software License 2.0.txt
org.mvel,mvel2,2.2.8.Final,Apache Software License 2.0.txt
org.sonatype.sisu.inject,guice-servlet,3.2.3,Eclipse Public License 1.0.txt
org.eclipse.sisu,org.eclipse.sisu.plexus,0.3.0.M1,Eclipse Public License 1.0.txt
org.eclipse.sisu,org.eclipse.sisu.inject,0.3.0.M1,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-util,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-impl,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-transport-wagon,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-connector-basic,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-transport-file,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-api,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-spi,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-transport-http,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.jdt.core.compiler,ecj,4.4.2,Eclipse Public License 1.0.txt
org.sonatype.sisu,sisu-guice,3.2.3,Eclipse Public License 1.0.txt
org.sonatype.plexus,plexus-cipher,1.7,Apache Software License 2.0.txt
org.sonatype.plexus,plexus-sec-dispatcher,1.3,Apache Software License 2.0.txt
org.antlr,antlr-runtime,3.5,BSD 3-clause New or Revised License.txt
com.thoughtworks.xstream,xstream,1.4.9,BSD 3-clause New or Revised License.txt
jQuery,MIT License.txt
AngularJS,MIT License.txt
angular-translate,MIT License.txt
ui-select2,MIT License.txt
angular-ui-bootstrap,MIT License.txt
Angular Treeview,MIT License.txt
Select2,Apache Software License 2.0.txt
Select2,GNU General Public License v2.0 only.txt
FileSaver.js,MIT License.txt
angular-file-upload,MIT License.txt
UI.Ace,MIT License.txt
Ace Code Editor,BSD 3-clause New or Revised License.txt
Font Awesome (Font),SIL Open Font License 1.1.txt
Glyphicons Halflings,MIT License.txt
Patternfly,Apache Software License 2.0.txt
OpenSans,Apache Software License 2.0.txt
Zocial,MIT License.txt
==> Normalizing license line endings
dos2unix: converting file rh-sso/org.apache.maven,maven-aether-provider,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.sonatype.plexus,plexus-cipher,1.7,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/angular-ui-bootstrap,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-compat,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-transport-wagon,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.ant,ant,1.8.3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-transport-file,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/Patternfly,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.codehaus.plexus,plexus-component-annotations,1.5.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/ui-select2,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven.wagon,wagon-http-shared,2.6,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/UI.Ace,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-spi,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-model,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-plugin-api,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.jdt.core.compiler,ecj,4.4.2,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/OpenSans,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-connector-basic,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/Font Awesome (Font),SIL Open Font License 1.1.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-transport-http,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.sonatype.sisu,sisu-guice,3.2.3,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven.wagon,wagon-http,2.6,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.google.code.guice,guice,2.0,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/aopalliance,aopalliance,1.0,Public Domain.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-repository-metadata,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/Select2,GNU General Public License v2.0 only.txt to Unix format ...
dos2unix: converting file rh-sso/org.codehaus.plexus,plexus-utils,3.0.20,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.antlr,antlr-runtime,3.5,BSD 3-clause New or Revised License.txt to Unix format ...
dos2unix: converting file rh-sso/com.google.zxing,core,3.2.1.redhat-4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-settings,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-model-builder,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.mvel,mvel2,2.2.8.Final,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/jQuery,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/Select2,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/Ace Code Editor,BSD 3-clause New or Revised License.txt to Unix format ...
dos2unix: converting file rh-sso/com.thoughtworks.xstream,xstream,1.4.9,BSD 3-clause New or Revised License.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-api,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.freemarker,freemarker,2.3.23.redhat-3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-settings-builder,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/angular-file-upload,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/Glyphicons Halflings,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/Angular Treeview,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.sisu,org.eclipse.sisu.plexus,0.3.0.M1,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/AngularJS,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.ant,ant-launcher,1.8.3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.codehaus.plexus,plexus-classworlds,2.5.2,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/Zocial,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.twitter4j,twitter4j-core,4.0.4.redhat-3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/angular-translate,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven.wagon,wagon-provider-api,2.6,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.codehaus.plexus,plexus-interpolation,1.21,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.sisu,org.eclipse.sisu.inject,0.3.0.M1,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/com.google.zxing,javase,3.2.1.redhat-4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.liquibase,liquibase-core,3.4.1.redhat-2,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-impl,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.sonatype.plexus,plexus-sec-dispatcher,1.3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/FileSaver.js,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.sonatype.sisu.inject,guice-servlet,3.2.3,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-artifact,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.eclipse.aether,aether-util,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file rh-sso/org.apache.maven,maven-core,3.2.5,Apache Software License 2.0.txt to Unix format ...
==> Symlinking identical files
Ace Code Editor,BSD 3-clause New or Revised License.txt
Angular Treeview,MIT License.txt
AngularJS,MIT License.txt
FileSaver.js,MIT License.txt
Font Awesome (Font),SIL Open Font License 1.1.txt
Glyphicons Halflings,MIT License.txt
OpenSans,Apache Software License 2.0.txt
 -> com.google.code.guice,guice,2.0,Apache Software License 2.0.txt
 -> org.apache.maven,maven-aether-provider,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-artifact,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-compat,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-core,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-model,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-model-builder,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-plugin-api,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-repository-metadata,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-settings,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-settings-builder,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven.wagon,wagon-http,2.6,Apache Software License 2.0.txt
 -> org.apache.maven.wagon,wagon-http-shared,2.6,Apache Software License 2.0.txt
 -> org.apache.maven.wagon,wagon-provider-api,2.6,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-classworlds,2.5.2,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-component-annotations,1.5.5,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-interpolation,1.21,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-utils,3.0.20,Apache Software License 2.0.txt
 -> org.liquibase,liquibase-core,3.4.1.redhat-2,Apache Software License 2.0.txt
 -> org.sonatype.plexus,plexus-cipher,1.7,Apache Software License 2.0.txt
 -> org.sonatype.plexus,plexus-sec-dispatcher,1.3,Apache Software License 2.0.txt
Patternfly,Apache Software License 2.0.txt
Select2,Apache Software License 2.0.txt
 -> Select2,GNU General Public License v2.0 only.txt
UI.Ace,MIT License.txt
 -> ui-select2,MIT License.txt
Zocial,MIT License.txt
angular-file-upload,MIT License.txt
angular-translate,MIT License.txt
angular-ui-bootstrap,MIT License.txt
aopalliance,aopalliance,1.0,Public Domain.txt
com.google.zxing,core,3.2.1.redhat-4,Apache Software License 2.0.txt
 -> com.google.zxing,javase,3.2.1.redhat-4,Apache Software License 2.0.txt
com.thoughtworks.xstream,xstream,1.4.9,BSD 3-clause New or Revised License.txt
jQuery,MIT License.txt
org.antlr,antlr-runtime,3.5,BSD 3-clause New or Revised License.txt
org.apache.ant,ant,1.8.3,Apache Software License 2.0.txt
 -> org.apache.ant,ant-launcher,1.8.3,Apache Software License 2.0.txt
org.eclipse.aether,aether-api,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-connector-basic,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-impl,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-spi,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-transport-file,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-transport-http,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-transport-wagon,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-util,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.jdt.core.compiler,ecj,4.4.2,Eclipse Public License 1.0.txt
org.eclipse.sisu,org.eclipse.sisu.inject,0.3.0.M1,Eclipse Public License 1.0.txt
 -> org.eclipse.sisu,org.eclipse.sisu.plexus,0.3.0.M1,Eclipse Public License 1.0.txt
 -> org.sonatype.sisu,sisu-guice,3.2.3,Eclipse Public License 1.0.txt
 -> org.sonatype.sisu.inject,guice-servlet,3.2.3,Eclipse Public License 1.0.txt
org.freemarker,freemarker,2.3.23.redhat-3,Apache Software License 2.0.txt
org.mvel,mvel2,2.2.8.Final,Apache Software License 2.0.txt
org.twitter4j,twitter4j-core,4.0.4.redhat-3,Apache Software License 2.0.txt
==> Complete
==> Deleting old license files in keycloak
==> Munging keycloak/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for keycloak/licenses.xml into keycloak
com.google.zxing,core,3.2.1,Apache Software License 2.0.txt
com.google.zxing,javase,3.2.1,Apache Software License 2.0.txt
org.liquibase,liquibase-core,3.4.1,Apache Software License 2.0.txt
org.twitter4j,twitter4j-core,4.0.4,Apache Software License 2.0.txt
org.freemarker,freemarker,2.3.23,Apache Software License 2.0.txt
aopalliance,aopalliance,1.0,Public Domain.txt
com.google.code.guice,guice,2.0,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-classworlds,2.5.2,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-utils,3.0.20,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-component-annotations,1.5.5,Apache Software License 2.0.txt
org.codehaus.plexus,plexus-interpolation,1.21,Apache Software License 2.0.txt
org.apache.ant,ant-launcher,1.8.3,Apache Software License 2.0.txt
org.apache.ant,ant,1.8.3,Apache Software License 2.0.txt
org.apache.maven.wagon,wagon-http-shared,2.6,Apache Software License 2.0.txt
org.apache.maven.wagon,wagon-provider-api,2.6,Apache Software License 2.0.txt
org.apache.maven.wagon,wagon-http,2.6,Apache Software License 2.0.txt
org.apache.maven,maven-compat,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-model-builder,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-core,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-model,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-artifact,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-plugin-api,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-settings,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-repository-metadata,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-aether-provider,3.2.5,Apache Software License 2.0.txt
org.apache.maven,maven-settings-builder,3.2.5,Apache Software License 2.0.txt
org.mvel,mvel2,2.2.8.Final,Apache Software License 2.0.txt
org.sonatype.sisu.inject,guice-servlet,3.2.3,Eclipse Public License 1.0.txt
org.eclipse.sisu,org.eclipse.sisu.plexus,0.3.0.M1,Eclipse Public License 1.0.txt
org.eclipse.sisu,org.eclipse.sisu.inject,0.3.0.M1,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-util,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-impl,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-transport-wagon,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-connector-basic,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-transport-file,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-api,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-spi,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.aether,aether-transport-http,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.jdt.core.compiler,ecj,4.4.2,Eclipse Public License 1.0.txt
org.sonatype.sisu,sisu-guice,3.2.3,Eclipse Public License 1.0.txt
org.sonatype.plexus,plexus-cipher,1.7,Apache Software License 2.0.txt
org.sonatype.plexus,plexus-sec-dispatcher,1.3,Apache Software License 2.0.txt
org.antlr,antlr-runtime,3.5,BSD 3-clause New or Revised License.txt
com.thoughtworks.xstream,xstream,1.4.9,BSD 3-clause New or Revised License.txt
jQuery,MIT License.txt
AngularJS,MIT License.txt
angular-translate,MIT License.txt
ui-select2,MIT License.txt
angular-ui-bootstrap,MIT License.txt
Angular Treeview,MIT License.txt
Select2,Apache Software License 2.0.txt
Select2,GNU General Public License v2.0 only.txt
FileSaver.js,MIT License.txt
angular-file-upload,MIT License.txt
UI.Ace,MIT License.txt
Ace Code Editor,BSD 3-clause New or Revised License.txt
Font Awesome (Font),SIL Open Font License 1.1.txt
Glyphicons Halflings,MIT License.txt
Patternfly,Apache Software License 2.0.txt
OpenSans,Apache Software License 2.0.txt
Zocial,MIT License.txt
==> Normalizing license line endings
dos2unix: converting file keycloak/org.liquibase,liquibase-core,3.4.1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-aether-provider,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.sonatype.plexus,plexus-cipher,1.7,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/angular-ui-bootstrap,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-compat,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-transport-wagon,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.ant,ant,1.8.3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-transport-file,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/Patternfly,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.codehaus.plexus,plexus-component-annotations,1.5.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/ui-select2,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven.wagon,wagon-http-shared,2.6,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/UI.Ace,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-spi,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/com.google.zxing,core,3.2.1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-model,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-plugin-api,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.jdt.core.compiler,ecj,4.4.2,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/OpenSans,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-connector-basic,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/Font Awesome (Font),SIL Open Font License 1.1.txt to Unix format ...
dos2unix: converting file keycloak/org.freemarker,freemarker,2.3.23,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-transport-http,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/org.sonatype.sisu,sisu-guice,3.2.3,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven.wagon,wagon-http,2.6,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/com.google.code.guice,guice,2.0,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.twitter4j,twitter4j-core,4.0.4,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/aopalliance,aopalliance,1.0,Public Domain.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-repository-metadata,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/Select2,GNU General Public License v2.0 only.txt to Unix format ...
dos2unix: converting file keycloak/org.codehaus.plexus,plexus-utils,3.0.20,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.antlr,antlr-runtime,3.5,BSD 3-clause New or Revised License.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-settings,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-model-builder,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.mvel,mvel2,2.2.8.Final,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/jQuery,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/Select2,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/Ace Code Editor,BSD 3-clause New or Revised License.txt to Unix format ...
dos2unix: converting file keycloak/com.thoughtworks.xstream,xstream,1.4.9,BSD 3-clause New or Revised License.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-api,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-settings-builder,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/angular-file-upload,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/Glyphicons Halflings,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/Angular Treeview,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.sisu,org.eclipse.sisu.plexus,0.3.0.M1,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/AngularJS,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.ant,ant-launcher,1.8.3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.codehaus.plexus,plexus-classworlds,2.5.2,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/Zocial,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/angular-translate,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven.wagon,wagon-provider-api,2.6,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.codehaus.plexus,plexus-interpolation,1.21,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.sisu,org.eclipse.sisu.inject,0.3.0.M1,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-impl,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/org.sonatype.plexus,plexus-sec-dispatcher,1.3,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/FileSaver.js,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.sonatype.sisu.inject,guice-servlet,3.2.3,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-artifact,3.2.5,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.eclipse.aether,aether-util,1.0.0.v20140518,Eclipse Public License 1.0.txt to Unix format ...
dos2unix: converting file keycloak/com.google.zxing,javase,3.2.1,Apache Software License 2.0.txt to Unix format ...
dos2unix: converting file keycloak/org.apache.maven,maven-core,3.2.5,Apache Software License 2.0.txt to Unix format ...
==> Symlinking identical files
Ace Code Editor,BSD 3-clause New or Revised License.txt
Angular Treeview,MIT License.txt
AngularJS,MIT License.txt
FileSaver.js,MIT License.txt
Font Awesome (Font),SIL Open Font License 1.1.txt
Glyphicons Halflings,MIT License.txt
OpenSans,Apache Software License 2.0.txt
 -> com.google.code.guice,guice,2.0,Apache Software License 2.0.txt
 -> org.apache.maven,maven-aether-provider,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-artifact,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-compat,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-core,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-model,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-model-builder,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-plugin-api,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-repository-metadata,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-settings,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven,maven-settings-builder,3.2.5,Apache Software License 2.0.txt
 -> org.apache.maven.wagon,wagon-http,2.6,Apache Software License 2.0.txt
 -> org.apache.maven.wagon,wagon-http-shared,2.6,Apache Software License 2.0.txt
 -> org.apache.maven.wagon,wagon-provider-api,2.6,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-classworlds,2.5.2,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-component-annotations,1.5.5,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-interpolation,1.21,Apache Software License 2.0.txt
 -> org.codehaus.plexus,plexus-utils,3.0.20,Apache Software License 2.0.txt
 -> org.liquibase,liquibase-core,3.4.1,Apache Software License 2.0.txt
 -> org.sonatype.plexus,plexus-cipher,1.7,Apache Software License 2.0.txt
 -> org.sonatype.plexus,plexus-sec-dispatcher,1.3,Apache Software License 2.0.txt
Patternfly,Apache Software License 2.0.txt
Select2,Apache Software License 2.0.txt
 -> Select2,GNU General Public License v2.0 only.txt
UI.Ace,MIT License.txt
 -> ui-select2,MIT License.txt
Zocial,MIT License.txt
angular-file-upload,MIT License.txt
angular-translate,MIT License.txt
angular-ui-bootstrap,MIT License.txt
aopalliance,aopalliance,1.0,Public Domain.txt
com.google.zxing,core,3.2.1,Apache Software License 2.0.txt
 -> com.google.zxing,javase,3.2.1,Apache Software License 2.0.txt
com.thoughtworks.xstream,xstream,1.4.9,BSD 3-clause New or Revised License.txt
jQuery,MIT License.txt
org.antlr,antlr-runtime,3.5,BSD 3-clause New or Revised License.txt
org.apache.ant,ant,1.8.3,Apache Software License 2.0.txt
 -> org.apache.ant,ant-launcher,1.8.3,Apache Software License 2.0.txt
org.eclipse.aether,aether-api,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-connector-basic,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-impl,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-spi,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-transport-file,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-transport-http,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-transport-wagon,1.0.0.v20140518,Eclipse Public License 1.0.txt
 -> org.eclipse.aether,aether-util,1.0.0.v20140518,Eclipse Public License 1.0.txt
org.eclipse.jdt.core.compiler,ecj,4.4.2,Eclipse Public License 1.0.txt
org.eclipse.sisu,org.eclipse.sisu.inject,0.3.0.M1,Eclipse Public License 1.0.txt
 -> org.eclipse.sisu,org.eclipse.sisu.plexus,0.3.0.M1,Eclipse Public License 1.0.txt
 -> org.sonatype.sisu,sisu-guice,3.2.3,Eclipse Public License 1.0.txt
 -> org.sonatype.sisu.inject,guice-servlet,3.2.3,Eclipse Public License 1.0.txt
org.freemarker,freemarker,2.3.23,Apache Software License 2.0.txt
org.mvel,mvel2,2.2.8.Final,Apache Software License 2.0.txt
org.twitter4j,twitter4j-core,4.0.4,Apache Software License 2.0.txt
==> Complete
/keycloak
/keycloak/distribution/saml-adapters/wildfly-adapter/wildfly-adapter-zip/src/main/resources/licenses /keycloak
==> Deleting old license files in rh-sso
==> Munging rh-sso/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for rh-sso/licenses.xml into rh-sso
==> Normalizing license line endings
==> Symlinking identical files
==> Complete
==> Deleting old license files in keycloak
==> Munging keycloak/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for keycloak/licenses.xml into keycloak
==> Normalizing license line endings
==> Symlinking identical files
==> Complete
/keycloak
/keycloak/distribution/saml-adapters/as7-eap6-adapter/eap6-adapter-zip/src/main/resources/licenses /keycloak
==> Deleting old license files in rh-sso
==> Munging rh-sso/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for rh-sso/licenses.xml into rh-sso
org.bouncycastle,bcpkix-jdk15on,1.52.0.redhat-3,MIT License.txt
org.bouncycastle,bcprov-jdk15on,1.52.0.redhat-3,MIT License.txt
==> Normalizing license line endings
dos2unix: converting file rh-sso/org.bouncycastle,bcprov-jdk15on,1.52.0.redhat-3,MIT License.txt to Unix format ...
dos2unix: converting file rh-sso/org.bouncycastle,bcpkix-jdk15on,1.52.0.redhat-3,MIT License.txt to Unix format ...
==> Symlinking identical files
org.bouncycastle,bcpkix-jdk15on,1.52.0.redhat-3,MIT License.txt
 -> org.bouncycastle,bcprov-jdk15on,1.52.0.redhat-3,MIT License.txt
==> Complete
==> Deleting old license files in keycloak
==> Munging keycloak/licenses.xml to ensure Windows filename compatibility
==> Downloading license files for keycloak/licenses.xml into keycloak
org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt
org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt
==> Normalizing license line endings
dos2unix: converting file keycloak/org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt to Unix format ...
dos2unix: converting file keycloak/org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt to Unix format ...
==> Symlinking identical files
org.bouncycastle,bcpkix-jdk15on,1.56,MIT License.txt
 -> org.bouncycastle,bcprov-jdk15on,1.56,MIT License.txt
==> Complete
root@ba09ed4de475:/keycloak# git diff
root@ba09ed4de475:/keycloak# git status
On branch KEYCLOAK-4950_master
Your branch is up-to-date with 'origin/KEYCLOAK-4950_master'.
nothing to commit, working directory clean
```

ASzc commented:
Also unable to reproduce with a VM of Ubuntu 16.04.

ASzc commented:
Could you post the output of these commands? They could make a difference:
```
root@vm:~/keycloak# uname -a
Linux vm 4.10.0-28-generic #32~16.04.2-Ubuntu SMP Thu Jul 20 10:19:48 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
root@vm:~/keycloak# locale
LANG=en_CA.UTF-8
LANGUAGE=en_CA:en
LC_CTYPE="en_CA.UTF-8"
LC_NUMERIC="en_CA.UTF-8"
LC_TIME="en_CA.UTF-8"
LC_COLLATE="en_CA.UTF-8"
LC_MONETARY="en_CA.UTF-8"
LC_MESSAGES="en_CA.UTF-8"
LC_PAPER="en_CA.UTF-8"
LC_NAME="en_CA.UTF-8"
LC_ADDRESS="en_CA.UTF-8"
LC_TELEPHONE="en_CA.UTF-8"
LC_MEASUREMENT="en_CA.UTF-8"
LC_IDENTIFICATION="en_CA.UTF-8"
LC_ALL=
root@vm:~/keycloak# df -Th .
Filesystem     Type  Size  Used Avail Use% Mounted on
/dev/vda1      ext4   13G  4.3G  7.8G  36% /
```
I think I'll try a new VM with a Norwegian locale, see if that makes a difference.

ASzc commented:
No difference
```
root@vm:~/keycloak# locale
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=nb_NO.UTF-8
LC_TIME=nb_NO.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=nb_NO.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=nb_NO.UTF-8
LC_NAME=nb_NO.UTF-8
LC_ADDRESS=nb_NO.UTF-8
LC_TELEPHONE=nb_NO.UTF-8
LC_MEASUREMENT=nb_NO.UTF-8
LC_IDENTIFICATION=nb_NO.UTF-8
LC_ALL=
```

stianst commented:
```
st@st:~/dev/keycloak$ uname -a
```

```
Linux st 4.10.0-35-generic #39-Ubuntu SMP Wed Sep 13 07:46:59 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
st@st:~/dev/keycloak$ locale
LANG=en_US.UTF-8
LANGUAGE=en_US
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=en_GB.UTF-8
LC_TIME=en_GB.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=en_GB.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=en_GB.UTF-8
LC_NAME=en_GB.UTF-8
LC_ADDRESS=en_GB.UTF-8
LC_TELEPHONE=en_GB.UTF-8
LC_MEASUREMENT=en_GB.UTF-8
LC_IDENTIFICATION=en_GB.UTF-8
LC_ALL=
st@st:~/dev/keycloak$ 
```

stianst commented:
I can't reproduce it now. Did you change anything?

stianst commented:
Here's the command I've used BTW:
````
find -name licenses.xml | grep -v target | xargs -I {} ./distribution/feature-packs/server-feature-pack/src/main/resources/licenses/common/download-license-files.sh {}
````

ASzc commented:
No changes since 00aea59 's force push, on Wednesday. Your commit stianst/keycloak@20b3bac has this as an ancestor commit, it was the first thing I checked.

stianst commented:
Ok, I'm happy with this then. @pdrozd could you take a look before we merge this?

pdrozd commented:
@ASzc The PR looks good to me, keycloak versions were updated, but I still see the problem with symlinks in the server-dist. There are no symlinks, but instead of them the files containing the license name:

```
ll keycloak-3.4.0.CR1-SNAPSHOT/docs/licenses-keycloak/
-rw-rw-r--. 1 pdrozd pdrozd 15289 17. říj 08.27 org.apache.ant,ant,1.8.3,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-aether-provider,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-artifact,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-compat,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-core,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-model-builder,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-model,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-plugin-api,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-repository-metadata,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-settings-builder,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven,maven-settings,3.2.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven.wagon,wagon-http-shared,2.6,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven.wagon,wagon-http,2.6,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.apache.maven.wagon,wagon-provider-api,2.6,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.codehaus.plexus,plexus-classworlds,2.5.2,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.codehaus.plexus,plexus-component-annotations,1.5.5,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.codehaus.plexus,plexus-interpolation,1.21,Apache Software License 2.0.txt
-rw-rw-r--. 1 pdrozd pdrozd    40 17. říj 08.27 org.codehaus.plexus,plexus-utils,3.0.20,Apache Software License 2.0.txt
```

The adapters dist looks ok.

ASzc commented:
Ah, probably missed the workaround for . For the record, the issue I filed with the wildfly plugin is [WFBUILD-27](https://issues.jboss.org/browse/WFBUILD-27)

ASzc committed:
Extract the license files with maven-dependency-plugin instead of wildfly-server-provisioning-maven-plugin

This works around WFBUILD-27 aka wildfly/wildfly-build-tools#39. Once that's merged and new version is released this commit can be reverted.

ASzc commented:
Added the workaround in a separate commit for easy reverting after wildfly/wildfly-build-tools#39 is merged.
